### ISSUE TITLE
Repo path issues, version updates

### ISSUE THREAD

**Issue by abuettner93** [caddy-dns/netlify#3]
Hey there!
It seems this module is using old repo paths for caddy (github.com/caddyserver/caddy/v2, where the v2 is no longer present). 
If its not too much to ask, could you update this module for the new structure of the caddy repo?

Im not a Go developer, so im a little lost on the updates required. Namely getting the `go.sum` components updated to reflect the new version (2.8.4).

Thanks!

**Comment by abuettner93** [comment]
Did more digging into Go dev - I see how things work now. But I still cant build and publish, so that might be something you have to do :)

**Comment by mholt** [comment]
What issue are you having, exactly?

**Comment by abuettner93** [comment]
There seems to be an issue with the formatting of the API call to LetsEncrypt - might be a version compatibility issue; not sure. I just figured if this was built for the newer version of Caddy, maybe that would help. I'll look into again when I get back to my server, but wanted to see if a new release for caddy 2.8.4 would be doable.

**Comment by abuettner93** [comment]
I just grabbed the error from my caddy logs:

ERR ts=1737487001.8020391 logger=tls.obtain msg=could not get certificate from issuer identifier=homelab.mydomain.com issuer=acme-v02.api.letsencrypt.org-directory error=[homelab.mydomain.com] solving challenges: presenting for challenge: adding temporary record for zone "mydomain.com.": json: cannot unmarshal object into Go struct field netlifyZone.domain of type string (order=https://acme-staging-v02.api.letsencrypt.org/acme/order/REDACTED) (ca=https://acme-staging-v02.api.letsencrypt.org/directory)


The information in the redacted URL is:

{
  "status": "invalid",
  "expires": "2025-01-28T19:16:41Z",
  "identifiers": [
    {
      "type": "dns",
      "value": "homelab.mydomain.com"
    }
  ],
  "authorizations": [
    "https://acme-staging-v02.api.letsencrypt.org/acme/authz/REDACTED"
  ],
  "finalize": "https://acme-staging-v02.api.letsencrypt.org/acme/finalize/REDACTED"
}


And the information in the "finalize" line is:

{
  "type": "urn:ietf:params:acme:error:malformed",
  "detail": "Method not allowed",
  "status": 405
}


Some cursory googling is telling me that certmagic may need to be updated in this plugin, but you would know best considering you wrote it haha.

**Comment by mholt** [comment]
> And the information in the "finalize" line is:
> 
> {
> "type": "urn:ietf:params:acme:error:malformed",
> "detail": "Method not allowed",
> "status": 405
> }

This is because you can't just `GET` that resource, you have to do a `POST` (or GET-as-POST for some other endpoints) so that's expected. This is a red herring.

The secondary issue is `json: cannot unmarshal object into Go struct field netlifyZone.domain of type string ` -- the struct type in the code doesn't match what the API is returning as JSON, so that'll need to be fixed here, then the library maintainer can get more insights into the error.

The real issue is still unknown since the error isn't being decoded and printed.

Anyway, I don't think they are related to your module version question originally.

**Comment by abuettner93** [comment]
Ok, so are you saying there is a fix/modification in this plugin that needs to change to better understand the error being received? And/or is the issue in the netlify API component? Sorry for the (dumb) questions; I know there are a number of moving parts here.

**Comment by mholt** [comment]
Yes; the code at libdns/netlify likely needs to be patched to get the error message, and then patched again to fix the error.

**Comment by abuettner93** [comment]
I have opened an issue with libdns/netlify maintainer and hope he will be able to assist.  [https://github.com/libdns/netlify/issues/5](https://github.com/libdns/netlify/issues/5)
